src/puppetlabs/trapperkeeper/authorization/rules.clj

```diff
@@ -170,7 +163,8 @@
   [rules :- Rules
    request :- ring/Request
    name :- schema/Str]
-  (if-let [ { matched-rule :rule matches :matches } (some #(match? % request) rules)]
+  (if-let [{matched-rule :rule matches :matches}
+           (some #(match? % request) (sort-rules rules))]
```

Comment:
Hm, that sounds like conflating too many things, and I don't want schema to be responsible for any of it.  I'm not convinced that's better, but feel free to file a JIRA and we can come back to it.